src/main/java/uk/gov/hmcts/reform/slc/model/FtpConfigProperties.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.hmcts.reform.slc.model;
+
+public class FtpConfigProperties {
+
+    public final String hostname;
+    public final int port;
+    public final String fingerprint;
+    public final String username;
+    public final String publicKey;
+    public final String privateKey;
+    public final String targetFolder;
+    public final String reportsFolder;
+
+    private FtpConfigProperties(Builder builder) {
+        this.hostname = builder.hostname;
+        this.port = builder.port;
+        this.fingerprint = builder.fingerprint;
+        this.username = builder.username;
+        this.publicKey = builder.publicKey;
+        this.privateKey = builder.privateKey;
+        this.targetFolder = builder.targetFolder;
+        this.reportsFolder = builder.reportsFolder;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static final class Builder {
+        private String hostname;
+        private int port;
+        private String fingerprint;
+        private String username;
+        private String publicKey;
+        private String privateKey;
+        private String targetFolder;
+        private String reportsFolder;
+
+        private Builder() {
+        }
+
+        public FtpConfigProperties build() {
+            return new FtpConfigProperties(this);
+        }
+
+        public Builder hostname(String hostname) {
+            this.hostname = hostname;
+            return this;
+        }
+
+        public Builder port(int port) {
+            this.port = port;
+            return this;
+        }
+
+        public Builder fingerprint(String fingerprint) {
+            this.fingerprint = fingerprint;
+            return this;
+        }
+
+        public Builder username(String username) {
+            this.username = username;
+            return this;
+        }
+
+        public Builder publicKey(String publicKey) {
+            this.publicKey = publicKey;
+            return this;
+        }
+
+        public Builder privateKey(String privateKey) {
+            this.privateKey = privateKey;
+            return this;
+        }
+
+        public Builder targetFolder(String targetFolder) {
+            this.targetFolder = targetFolder;
+            return this;
+        }
+
+        public Builder reportsFolder(String reportsFolder) {
+            this.reportsFolder = reportsFolder;
+            return this;
+        }
+    }
+}
```

Comment:
imho this only added complexity... :/

Reply:
to be honest I was not sure if this was the way to go to fix the problem.Feel free to suggest any better alternatives :)

Reply:
well the project is in a pretty good shape ;)
<img width="527" alt="screen shot 2018-03-01 at 11 45 43" src="https://user-images.githubusercontent.com/2472141/36843226-24cf1d06-1d46-11e8-8fd6-509386981807.png">

Reply:
Yes that should be fine .If everyone @doncem @lgonczar @KarolJastrzebski  is in consensus and fine can we just ignore this warning?

Reply:
the new version also violates Codacy rules

Reply:
I'm ok with it - sonar show us potential issues, but it's up to us to decide whether it's an actual problem. If the constructor accepted arguments of different categories, that would be different, but they're all FTP parameters.

I think your changes are fine, too, and would be perfectly happy to see them in master.